io/net/virt-net/network_virtualization.py

```diff
@@ -420,6 +423,36 @@ def test_vnic_auto_failover(self):
         else:
             self.cancel("Provide more backing device, only 1 given")
 
+    def test_rmdev_viosfailover(self):
+        '''
+        using mrdev and mkdev command to check vios failover works
+        '''
+        cmd = "lsmap -all -vnic -cpid %s" % self.lpar_id
+        vnic_server = self.run_command(cmd, self.con_vios)[2].split()[0]
+        cmd = "lsmap -vnic -vadapter %s" % vnic_server
+        output = self.run_command(cmd, self.con_vios)
+        for line in output:
+            if 'Backing device' in line:
+                vnic_backing_device = line.split(':')[-1]
+        before = self.get_active_device_logport(self.slot_num[0])
+        cmd = 'rmdev -l %s' % vnic_server
+        self.check_defined_available(cmd)
+        cmd = 'rmdev -l %s' % vnic_backing_device
+        self.check_defined_available(cmd)
+        after = self.get_active_device_logport(self.slot_num[0])
+        self.log.debug("Active backing device: %s", after)
+        if before == after:
+            self.fail("failover not occour")
+        time.sleep(20)
+        cmd = 'mkdev -l %s' % vnic_backing_device
+        self.check_defined_available(cmd)
+        cmd = 'mkdev -l %s' % vnic_server
+        self.check_defined_available(cmd)
```

Comment:
@smruti77 @abdhaleegit I have tried to simplify the code a little, please check if it looks fine.

```
    def test_rmdev_viosfailover(self):
        cmd = "lsmap -all -vnic -cpid %s" % self.lpar_id
        vnic_server = self.run_command(cmd, self.con_vios)[2].split()[0]

        cmd = "lsmap -vnic -vadapter %s" % vnic_server
        output = self.run_command(cmd, self.con_vios)

        vnic_backing_device = None
        for line in output:
            if 'Backing device' in line:
                vnic_backing_device = line.split(':')[-1]

        before = self.get_active_device_logport(self.slot_num[0])
        self.log.debug("Active backing device before : %s", before)

        self.validate_vios_command('rmdev -l %s' % vnic_server, 'Defined')
        if vnic_backing_device:
            self.validate_vios_command('rmdev -l %s' % vnic_backing_device, 'Defined')

        after = self.get_active_device_logport(self.slot_num[0])
        self.log.debug("Active backing device after: %s", after)

        if before == after:
            self.fail("failover not occour")
        time.sleep(20)

        if vnic_backing_device:
            self.validate_vios_command('mkdev -l %s' % vnic_backing_device, 'Available')
        self.validate_vios_command('mkdev -l %s' % vnic_server, 'Available')
```
```
    def validate_vios_command(self, cmd, validate_string):
        l_cmd = "echo \"%s\" | oem_setup_env" % cmd
        output = self.run_command(l_cmd, self.con_vios)
        if validate_string not in str(output):
            self.fail("command fail in vios")
        elif validate_string not in str(output):
            self.fail("command fail in vios")
```

Reply:
here one time is fine, others are looking good.
```
       if validate_string not in str(output):
            self.fail("command fail in vios")
        elif validate_string not in str(output):
            self.fail("command fail in vios")